include/fem-dem/gls_sharp_navier_stokes.h

```diff
@@ -362,10 +363,26 @@ class GLSSharpNavierStokesSolver : public GLSNavierStokesSolver<dim>
   cell_cut(const typename DoFHandler<dim>::active_cell_iterator &cell,
            std::vector<types::global_dof_index> &         local_dof_indices,
            std::map<types::global_dof_index, Point<dim>> &support_points);
+
+  /**
+   * @brief
+   * Return a bool to define if a cell is cut by an IB particle and the local
+   * DOFs of the cell for later use. This function is built to handle special
+   * cases where the level set is always positive.
+   *
+   * @param cell, the cell that we verify whether it is cut or not.
+   *
+   * @param support_points, a mapping of support points for the DOFs.
+   *
+   * @param p, the particle index of the particle used in the check
```

Comment:
```suggestion
   * @param p, the particle index of the particle used in the check
```
Question. Instead, could we pass a reference to the particle itself to this function? if so, we could implement the entire function outside of the gls sharp navier stokes class and it would be a lot cleaner than to have the class grow indefinitely like it does. This type of function should have unit test for them...